### PR TITLE
Issue #7523 -  Geocarousel improve markers viz

### PR DESCRIPTION
**Pull request by ale-cristofori**
## Description
This RP aims at putting into place improvements for the visualisation of marker items for the GeoCarousel layout, within the geostories functionality.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) N/A

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Please describe: Improvement request from customer

## Issue
**What is the current behavior?**

When items on the map were too close to each other it was rather difficult to distinguish the selected one from those that were not, this was due to two main issues:

1. The colour of the selected marker did not change, the selected one was added a 'chevron' icon above.
2. The selected item z index did not change and it was not drawn on top of the unselected markers, thus it was sometimes hidden by other markers making it difficult to see. 

**What is the new behavior?**
In response to the issues described above fix were applied as follows:

1. The colour of the selected item now is different then the non-selected ones, whenever an item is selected its colour changes to red.
2. The highlighted/selected items layer is now separated from the unselected/non-highlighted one. The highlighted markers layer is added on top the non-highlighted items one and its z index is higher, as a consequence the selected items will always be drawn on top of non-selected ones. See below

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/components/geostory/layouts/sections/GeoCarousel.jsx [hunk 1]
```diff
@@ -72,6 +73,10 @@ const GeoCarousel = ({
     defaultMarkerStyle = {
         iconColor: 'cyan',
         iconShape: 'circle'
+    },
+    highlightedMarkerStyle = {
+        iconColor: 'red',
```
Comment (offtherailz) [inline web/client/components/geostory/layouts/sections/GeoCarousel.jsx:78]:
```suggestion
        iconColor: 'green',
```
green is more understandable.
image

File: web/client/plugins/GeoStory.jsx [hunk 2]
```diff
@@ -146,6 +147,11 @@ GeoStory.defaultProps = {
  *           // -> 'purple', 'violet', 'pink', 'green-dark', 'green', 'green-light' or 'black'
  *       "iconShape": "square" // 'circle', 'square', 'star' or 'penta'
  *     },
+ *     "highlightedMarkerStyle": {
+ *       "iconColor": "red", // 'orange', 'orange-dark', 'orange', 'yellow', 'blue-dark', 'blue', 'cyan', ->
+ *           // -> 'purple', 'violet', 'pink', 'green-dark', 'green', 'green-light' or 'black'
```
Comment (offtherailz) [inline web/client/plugins/GeoStory.jsx:152]:
```suggestion
 *       "iconColor": "green", // 'orange', 'orange-dark', 'orange', 'yellow', 'blue-dark', 'blue', 'cyan', ->
 *           // -> 'purple', 'violet', 'pink', 'green-dark', 'red', 'green-light' or 'black'
```

File: web/client/utils/__tests__/GeoStoryUtils-test.js [hunk 3]
```diff
@@ -636,4 +637,52 @@ describe("GeoStory Utils", () => {
...
+        const content = {id: 'cd29a263-d36a-4459-8821-8dce4c761bcc'};
+        const contentId = 'cd29a263-d36a-4459-8821-8dce4c761bcc';
+        const feature = {
+            type: 'Feature',
+            geometry: {},
+            properties: {}
+        };
+        const markerStyle = getContentsFeatureStyle(defaultMarkerStyle, contents, content, contentId, feature);
+        expect(markerStyle).toEqual({
+            iconColor: 'red',
```
Comment (offtherailz) [inline web/client/utils/__tests__/GeoStoryUtils-test.js:682]:
```suggestion
            iconColor: 'green',
            iconShape: 'circle'
        };
        const contents = [
            {id: 'cd29a263-d36a-4459-8821-8dce4c761bcc'},
            {id: 'ef332a5b-3405-4f33-b470-5673cc146fea'}
        ];
        const content = {id: 'cd29a263-d36a-4459-8821-8dce4c761bcc'};
        const contentId = 'cd29a263-d36a-4459-8821-8dce4c761bcc';
        const feature = {
            type: 'Feature',
            geometry: {},
            properties: {}
        };
        const markerStyle = getContentsFeatureStyle(defaultMarkerStyle, contents, content, contentId, feature);
        expect(markerStyle).toEqual({
            iconColor: 'green',
```

File: web/client/utils/GeoStoryUtils.js [hunk 4]
```diff
@@ -656,3 +656,10 @@ export function getVectorLayerFromContents({
         ], []).reverse()
     };
 }
+
```
Comment (offtherailz) [inline web/client/utils/GeoStoryUtils.js:659]:
```suggestion

/**
 * Applies to the contents style passed the iconText (1, 2, 3), merges the default marker style with the specific feature style and adds the `highlighted` flag.
 */
```

**Review by offtherailz — CHANGES_REQUESTED** [review]
Looks good, I only changed the style of the highlighted marker from `red` to `green`. I think is more quite color to show the current active. Red has a warning effect. 
I think you can apply my suggestions inline and re-request my review, so we can merge

**Comment by ale-cristofori** [comment]
@offtherailz, thanks for the suggestions, I applied the changes using the inline commits, is now ready to be re-reviewed.

**Review by offtherailz — APPROVED** [review]

**Comment by offtherailz** [comment]
@ElenaGallo, could you please test this on  DEV  ? Thank you

**Comment by ElenaGallo** [comment]
Test passed, @ale-cristofori please backport to 2021.02.xx. Thanks

**Pull request merged**